lib/record_store/zone.rb

```diff
@@ -21,19 +21,74 @@ def each(&block)
 
       def find(name)
         return unless File.exists?(zone_path = "#{RecordStore.zones_path}/#{name}.yml")
-        Zone.from_yaml_definition(*YAML.load_file(zone_path).first)
+        load_yml_zone_definition(zone_path).first.last
+      end
+
+      def write(name, config:, records:, format: :file)
+        raise ArgumentError, "format must be :directory or :file" unless %i(file directory).include?(format)
+        name = name.chomp('.')
+        zone_file = "#{RecordStore.zones_path}/#{name}.yml"
+        zone = { name => { config: config.to_hash } }
+        records = records.map(&:to_hash).sort_by! {|r| [r.fetch(:fqdn), r.fetch(:type), r[:nsdname] || r[:address]]}
+
+        zone[name][:records] = records if format == :file
```

Comment:
I did that in bbf375156d23fa3a52f245a0eebc1f122485cf3d but I'm not sure I want to keep it, because some of the logic is shared.
```ruby
if format == :file
  zone[name][:records] = records
  write_yml_file(zone_file, zone.deep_stringify_keys)
  remove_record_files(name)
else
  write_yml_file(zone_file, zone.deep_stringify_keys)
  remove_record_files(name)
  write_record_files(name, records)
end
```